motionplan/nearestNeighbor.go

```diff
@@ -43,14 +43,14 @@ func kNearestNeighbors(planOpts *PlannerOptions, rrtMap map[node]node, target []
 		allCosts = append(allCosts, &neighbor{dist: dist, node: node})
 	}
 	sort.Slice(allCosts, func(i, j int) bool {
-		return allCosts[i].dist < allCosts[j].dist
+		return (allCosts[i].dist + allCosts[i].node.(*costNode).cost) < (allCosts[j].dist + allCosts[j].node.(*costNode).cost)
```

Comment:
[minor] this doesn't seem right, since you could call this on a map without costNodes

Reply:
That's true; this should check that this is a costnode.